### ISSUE TITLE
org.cactoos.concurrent.* EO primitives are required

### ISSUE THREAD

**Issue by dgroup** [yegor256/cactoos#962]
I couldn't find any EO primitives/objects related to `java.util.concurrent` like:
 - Timeout (with unit/value)
 - Fake implementations/envelopes at least for ThreadFactory, ExecutorService/Executor.

This feature shouldn't affect the classes like `org.cactoos.scalar.SyncScalar` and they should be kept at the same packages as they exist now.

Let's consider the possibility of the implementation of these features and the architecture approach here.

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by llorllale** [comment]
@dgroup 

> Timeout (with unit/value)

Have you looked at `TimedFunc`?

> Fake implementations/envelopes at least for ThreadFactory, ExecutorService/Executor.

`cactoos` doesn't do fakes. What do you propose for `ThreadFactory, ExecutorService/Executor`?

**Comment by dgroup** [comment]
@llorllale , 
> TimedFunc
Yes, i did. The timed function has only one unit - milliseconds. I'm proposing something like 
```java
interface Timeout {

    Long timeout();
    TimeUnit measure();

    class No implements Timeout {

        @Override
        public Long timeout() {
            return 0L;
        }

        @Override
        public TimeUnit measure() {
            return TimeUnit.MILLISECONDS;
        }
    }
}
```
with different units.
> `cactoos` doesn't do fakes. What do you propose for ThreadFactory, ExecutorService/Executor?

Not just fakes, also envelopes like `org.cactoos.list.ListEnvelope` for ThreadFactory, ExecutorService.

**Comment by llorllale** [comment]
@dgroup have you considered [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) instead of your `Timeout` proposal?

> envelopes like org.cactoos.list.ListEnvelope for ThreadFactory, ExecutorService

Do we really need to? Can't we just re-imagine things and design our own pooling. Maybe a class/interface called `Threads` or something like that.

**Comment by dgroup** [comment]
@llorllale, thanks, the Duration is good alternative, i wasn't aware of it.
I like your idea with `Threads`. For example smth  like
```java
... class Threads implements Iterable<Future> {
  ...
}
```
The constructors may consume the Scalar and Callable/Runnable (for compatibility).

**Comment by llorllale** [comment]
@dgroup I think we should create a new package `experimental` and add this one there. I'm not sure I want to commit this to the fully supported API.

**Comment by dgroup** [comment]
@llorllale, agree.

**Comment by llorllale** [comment]
@0crat assign @dgroup

**Comment by llorllale** [comment]
@dgroup let's place it inside the new `experimental` package then

**Comment by 0crat** [comment]
> @0crat assign @dgroup  ([here](https://github.com/yegor256/cactoos/issues/962#issuecomment-442462698))

@llorllale The job [#962](https://github.com/yegor256/cactoos/issues/962) assigned to @dgroup[/z](https://www.0crat.com/u/dgroup), here is [why](https://www.0crat.com/footprint/C63314D6Z/d11ed8d3-f961-4ad1-ada5-83b8f7cd026c); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by llorllale** [comment]
@0crat assign @dgroup

**Comment by 0crat** [comment]
> @0crat assign @dgroup ([here](https://github.com/yegor256/cactoos/issues/962#issuecomment-445413246))

@llorllale Job [#962](https://github.com/yegor256/cactoos/issues/962) is already assigned to @dgroup[/z](https://www.0crat.com/u/dgroup)

**Comment by llorllale** [comment]
@dgroup I think we can close this

**Comment by 0crat** [comment]
The job [#962](https://github.com/yegor256/cactoos/issues/962) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @dgroup[/z](https://www.0crat.com/u/dgroup)

**Comment by 0pdd** [comment]
@dgroup the puzzle [#1000](https://github.com/yegor256/cactoos/issues/1000) is still not solved; solved: [#972](https://github.com/yegor256/cactoos/issues/972).

**Comment by 0pdd** [comment]
@dgroup the puzzle [#1277](https://github.com/yegor256/cactoos/issues/1277) is still not solved; solved: [#1000](https://github.com/yegor256/cactoos/issues/1000), [#972](https://github.com/yegor256/cactoos/issues/972).

**Comment by 0pdd** [comment]
@dgroup 2 puzzles [#1406](https://github.com/yegor256/cactoos/issues/1406), [#1407](https://github.com/yegor256/cactoos/issues/1407) are still not solved; solved: [#1000](https://github.com/yegor256/cactoos/issues/1000), [#1277](https://github.com/yegor256/cactoos/issues/1277), [#972](https://github.com/yegor256/cactoos/issues/972).

**Comment by 0pdd** [comment]
@dgroup the puzzle [#1406](https://github.com/yegor256/cactoos/issues/1406) is still not solved; solved: [#1000](https://github.com/yegor256/cactoos/issues/1000), [#1277](https://github.com/yegor256/cactoos/issues/1277), [#1407](https://github.com/yegor256/cactoos/issues/1407), [#972](https://github.com/yegor256/cactoos/issues/972).

**Comment by 0pdd** [comment]
@dgroup all 5 puzzles are solved here: [#1000](https://github.com/yegor256/cactoos/issues/1000), [#1277](https://github.com/yegor256/cactoos/issues/1277), [#1406](https://github.com/yegor256/cactoos/issues/1406), [#1407](https://github.com/yegor256/cactoos/issues/1407), [#972](https://github.com/yegor256/cactoos/issues/972).
